lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -604,23 +604,26 @@ struct StringConstantOpConv : public OpConversionPattern<StringConstantOp> {
   LogicalResult
   matchAndRewrite(moore::StringConstantOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    const auto str = op.getValue();
-    const unsigned byteWidth = str.size() * 8;
     const auto resultType =
         typeConverter->convertType(op.getResult().getType());
-    if (const auto intType = mlir::dyn_cast<IntegerType>(resultType)) {
-      if (intType.getWidth() < byteWidth) {
-        return rewriter.notifyMatchFailure(op,
-                                           "invalid string constant type size");
-      }
-    } else {
+    const auto intType = mlir::dyn_cast<IntegerType>(resultType);
+    if (!intType) {
       return rewriter.notifyMatchFailure(op, "invalid string constant type");
     }
```

Comment:
Nit: I think this message is not needed (probably a holdover from earlier Moore dialect days), since the `StringConstantOp` verifier verifies that its result type is a `moore::IntType`. Instead of `dyn_cast`, a `cast` works :smiley: 

```suggestion
    const auto intType = mlir::cast<IntegerType>(resultType);
```

Reply:
Oh, sure, missed that. Thanks